notebook/static/notebook/js/outputarea.js

```diff
@@ -783,6 +787,18 @@ define([
         element.append(toinsert);
         return toinsert;
     };
+    
+    var append_mathml = function (mathml, md, element) {
+        /**
+         * This method cannot do the typesetting because the mathml first has to
+         * be on the page.
+         */
+        var type = 'application/mathml+xml';
+        var toinsert = this.create_output_subarea(md, "output_mathml", type);
+        toinsert.text(mathml);
```

Comment:
I think the key thing is to use the trust mechanism, as we do for HTML and JS output already. So outputs sent from a running kernel or loaded from a trusted notebook will be displayed without sanitisation, and outputs loaded from an *untrusted* notebook are either not displayed at all, or sanitised so they can't do anything dangerous.

IIRC the sanitisation library we use is Google Caja.